fog/ledger/server/src/db_fetcher.rs

```diff
@@ -0,0 +1,299 @@
+        let _ = self.stop();
+    }
+}
+
+struct DbFetcherThread<DB: Ledger, E: LedgerEnclaveProxy + Clone + Send + Sync + 'static> {
+    db: DB,
+    stop_requested: Arc<AtomicBool>,
+    next_block_index: u64,
+    logger: Logger,
+    enclave: E,
+    watcher: WatcherDB,
+    db_poll_shared_state: Arc<Mutex<DbPollSharedState>>,
+}
+
+/// Background worker thread implementation that takes care of periodically
+/// polling data out of the database. Add join handle
+impl<DB: Ledger, E: LedgerEnclaveProxy + Clone + Send + Sync + 'static> DbFetcherThread<DB, E> {
+    const POLLING_FREQUENCY: Duration = Duration::from_millis(10);
+    const ERROR_RETRY_FREQUENCY: Duration = Duration::from_millis(1000);
+
+    pub fn start(
+        db: DB,
+        stop_requested: Arc<AtomicBool>,
+        next_block_index: u64,
+        logger: Logger,
+        enclave: E,
+        watcher: WatcherDB,
+        db_poll_shared_state: Arc<Mutex<DbPollSharedState>>,
+    ) {
+        let thread = Self {
+            db,
+            stop_requested,
+            next_block_index,
+            logger,
+            enclave,
+            watcher,
+            db_poll_shared_state,
+        };
+        thread.run();
+    }
+
+    fn run(mut self) {
+        log::info!(self.logger, "Db fetcher thread started.");
+        self.next_block_index = 0;
+        loop {
+            if self.stop_requested.load(Ordering::SeqCst) {
+                log::info!(self.logger, "Db fetcher thread stop requested.");
+                break;
+            }
+            // Each call to load_block_data attempts to load one block for each known
+            // invocation. We want to keep loading blocks as long as we have data to load,
+            // but that could take some time which is why the loop is also gated
+            // on the stop trigger in case a stop is requested during loading.
+            while self.load_block_data() && !self.stop_requested.load(Ordering::SeqCst) {}
+        }
+    }
+
+    /// Attempt to load the next block that we
+    /// are aware of and tracking.
+    /// Returns true if we might have more block data to load.
+    fn load_block_data(&mut self) -> bool {
+        let mut has_more_work = false;
+        let mut records: Vec<KeyImageData> = Vec::new();
+        let watcher_timeout: Duration = Duration::from_millis(5000);
+
+        match self.db.get_block_contents(self.next_block_index) {
+            Err(LedgerError::NotFound) => std::thread::sleep(Self::POLLING_FREQUENCY),
+            Err(e) => {
+                log::error!(
+                    self.logger,
+                    "Unexpected error when checking for block data {}: {:?}",
+                    self.next_block_index,
+                    e
+                );
+                std::thread::sleep(Self::ERROR_RETRY_FREQUENCY);
+            }
+            Ok(block_contents) => {
+                // Get the timestamp for the block.
+                let mytimestamp = Self::get_watcher_timestamp(
```

Comment:
`mytimestamp` is not good because the style of our code is that we separate words with underscore

I think it would be better to just name this variable `timestamp` because there will be slightly less code later if we do

Reply:
okay will make change